server/src/api/api.ts

```diff
@@ -248,11 +305,131 @@ const resolvers: any = {
             ...args.updatedItem
         };
     },
-    categories: async (root, _args, _context): Promise<[Category]> => {
+    createRequest: async (root, _args, _context) => {
         // @ts-ignore
         const {args, context} = fixArguments(root, _args, _context);
-        return await DB.from("categories");
+        // if non-admin, user on request must be user signed in
+        // TODO: test this
+        if (!context.user.admin && context.user.uuid !== args.newRequest.user_id) {
+            throw new GraphQLError("Unable to create request because you are not an admin and your UUID " +
+                "does not match the UUID of the user this request is for");
+        }
+
+        // make sure the user the request is for exists
+        const users = await DB.from("users").where({
+            uuid: args.newRequest.user_id
+        }).select("name", "email", "uuid", "phone", "slackUsername", "haveID", "admin");
+
+        let user;
+        if (users.length === 0) {
+            throw new GraphQLError("Unable to create this request because no user with the UUID provided was found");
+        } else {
+            user = users[0];
+        }
+
+        // fetch the item
+        const item: any = await getItem(args.newRequest.request_item_id, context.user.admin);
+
+        if (!item) {
+            throw new GraphQLError("Can't create request for item that doesn't exist!  Item ID provided: ", args.newRequest.request_item_id);
+        }
+
+        // clip item quantity to allowed values
+        if (args.newRequest.quantity > item.maxRequestQty) {
```

Comment:
where's the quantity remaining check?

Reply:
Not implemented yet, will be done with the rest of quantity bits